hack/lib/util/golang.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+#
+# This library holds golang related utility functions.
+
+# os::golang::verify_go_version ensure the go tool exists and is a viable version.
+function os::golang::verify_go_version() {
+  if ! which go &>/dev/null; then
```

Comment:
You should make your mind, because all over our codebase we use 2 space, 4 spaces and tab interchangeably ;)